### PR TITLE
Adding argument for metadata indexing by generate_inputs

### PR DESCRIPTION
**Pull request by myousif9**
Adding argument for metadata indexing with pybids in `generate_inputs` function which allows metadata to be accessible in `BIDSLayout` returned from `<generate_inputs_return>.layout`.

### CODE REVIEW

**Review by pvandyken — CHANGES_REQUESTED** [review]
The repo will need to be formatted with `ruff`, `poetry run poe fix` should do the trick.

Otherwise, looks good

File: snakebids/tests/test_generate_inputs.py [hunk 1]
```diff
@@ -1,10 +1,14 @@
 # ruff: noqa: PLR2004
 from __future__ import annotations
+from distutils import extension
```
[the remark this answers is not in the document]
Comment (myousif9) [reply snakebids/tests/test_generate_inputs.py]: I'm not sure where they came from, I don't remember using those imports. Though it appears those new imports disappeared after running ruff formatting.

**Comment by pvandyken** [comment]
@myousif9 feel free to add an ignore comment to the ruff error you're getting

**Comment by myousif9** [comment]
Ran ruff formatting.

**Comment by myousif9** [comment]
Added ignore comment for ruff error, synced with main branch and now passes all tests.

**Review by kaitj — APPROVED** [review]

**Review by pvandyken — APPROVED** [review]

**Pull request merged**